samples/net/mqtt_publisher/src/main.c

```diff
@@ -61,15 +61,69 @@ struct mqtt_client_ctx {
 	void *publish_data;
 };
 
-/* This is the network context structure. */
-static struct net_context *net_ctx;
-
 /* The mqtt client struct */
 static struct mqtt_client_ctx client_ctx;
 
 /* This routine sets some basic properties for the network context variable */
-static int network_setup(struct net_context **net_ctx, const char *local_addr,
-			 const char *server_addr, u16_t server_port);
+static int network_setup(void);
+
+#if defined(CONFIG_MQTT_LIB_TLS)
+
+#include "test_certs.h"
+
+/* TLS */
+#define TLS_SNI_HOSTNAME "localhost"
+#define TLS_REQUEST_BUF_SIZE 1500
```

Comment:
I think MBEDTLS_SSL_MAX_CONTENT_LEN is related to the maximum size of SSL record, According to the SSL protocol, it can be 16384 at most. During the handshake with mosquitto, the SSL record is larger than 1500, this is why handshake failed.

TLS_REQUEST_BUF_SIZE is the buffer size we used to hold the data for mbedtls_ssl_read() and mbedtls_ssl_write(). But it should be less than MBEDTLS_SSL_MAX_CONTENT_LEN due to the protocol limitation.